### PR TITLE
Add action bar support

### PR DESCRIPTION
**Pull request by LeoDog896**
This adds an action bar packet + function, as well as integration into conway.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -485,6 +485,11 @@ impl<C: Config> Client<C> {
         }
     }
 
+    /// Sets the action bar for this client.
+    pub fn set_action_bar(&mut self, text: impl Into<Text>) {
+        self.send_packet(SendActionBar { text: text.into() });
```
Comment (rj00a) [inline src/client.rs:490]: Might be better to defer sending this packet.

The reason is that the login/join game packet is also deferred, and a lot of packets don't work correctly if they are sent before the login packet.

The other reason is that if set_action_bar is called multiple times in a single tick, you'll only send one packet. But then you have to add a field to `Client`.

But whatever this works fine. Just something to think about.

File: src/protocol/packets/s2c.rs [hunk 2]
```diff
@@ -629,6 +629,12 @@ pub mod play {
         }
     }
 
+    def_struct! {
+        SendActionBar {
```
Comment (rj00a) [inline src/protocol/packets/s2c.rs:633]: Should call this `OverlayMessage` for consistency with packets.json. I know it's dumb.

**Review by rj00a — CHANGES_REQUESTED** [review]
I think there's a merge conflict now.

File: src/protocol/packets/s2c.rs (hunk 2 above)
Comment (LeoDog896) [reply src/protocol/packets/s2c.rs:633]: It's not, though possibly for the def_struct macro for packets, I think using the names from JSON as a type hint / error would be good

**Pull request merged**